gooddata-java/src/main/java/com/gooddata/sdk/service/ValidatingResponseHandler.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2004-2021, GoodData(R) Corporation. All rights reserved.
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE.txt file in the root directory of this source tree.
+ */
+package com.gooddata.sdk.service;
+
+import com.github.sardine.impl.SardineException;
+import com.gooddata.sdk.service.gdc.GdcSardineException;
+import org.apache.http.Header;
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpStatus;
+import org.apache.http.StatusLine;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.ResponseHandler;
+
+import java.io.IOException;
+
+import static com.gooddata.sdk.common.gdc.Header.GDC_REQUEST_ID;
+
+/**
+ * A basic response handler checks if the status code is in the valid range. If not, it takes X-GDC-REQUEST header
+ * and set it on the stack trace.
+ */
+public class ValidatingResponseHandler implements ResponseHandler<Void> {
```

Comment:
Shouldn't be this package private class located within the `com.gooddata.sdk.service.gdc` package next to all "data store" (or "sardine") classes? Also the naming is quite confusing because as I can see in the code, it just handles (validates) Sardine exceptions.

Reply:
Thanx for comments. Changed to package-private and moved to `com.gooddata.sdk.service.gdc`. Regarding to the naming, I'm just name same as it in ` com.github.sardine.impl.handler.ValidatingResponseHandler`, because it's doing almost same thing. But if you have some better idea for a name, let me know, pls.

Reply:
Well, it could probably be `SardineResponseHandler` or `GdcSardineResponseHandler`. Also, if it's a copy of/inspiration from the original `com.github.sardine.impl.handler.ValidatingResponseHandler`, I would recommend mentioning that in the class level JavaDoc.

Reply:
Changed and added mention.